app/helpers/commonwealth_vlr_engine/catalog_helper_behavior.rb

```diff
@@ -62,15 +74,58 @@ def insert_opengraph_markup
     def link_to_az_value(letter, field, search_path, link_class = nil)
       new_params = params.permit!.except(:controller, :action, :q, :page)
       new_params[:q] = "#{field}:#{letter}*"
-      link_to(letter,
-              self.send(search_path, new_params),
-              class: link_class)
+      link_to(letter, self.send(search_path, new_params), class: link_class)
+    end
+
+    # @param document [SolrDocument]
+    # @return [Boolean]
+    def harvested_object?(document)
+      document[blacklight_config.hosting_status_field.to_sym] == 'harvested'
+    end
+
+    # @param document_files [Array] Curator::Filestreams::Document SolrDocument objects
+    # @return [Boolean]
+    def pdf_url_for_viewer(document_files)
+      pdf_file = document_files.find { |a| a['attachments_ss']['document_access'].present? }
+      filestream_disseminator_url(pdf_file['storage_key_base_ss'], 'document_access')
+    end
+
+    # @param document [SolrDocument]
+    # @param files_hash [Hash] output of CommonwealthVlrEngine::Finder.get_files
+    # @return [Boolean]
+    def render_image_viewer?(document, files_hash)
+      has_image_files?(files_hash) && files_hash[:image].length <= 7 && !has_searchable_text?(document)
+    end
+
+    def render_image_viewer(document, files_hash)
+      if files_hash[:image].length == 1
```

Comment:
i prefer using `.count` as opposed to `length` since its faster according to benchmarks I've seen

Reply:
Also you could do a `case` `when` here instead of an `if` `elsif` like so

```
case file_hash[:image].count
when 1
render partial...
when 2..7
render partial:...
```

Reply:
Refactored, see 4db414c.